RIOT-OS/RIOT
# cpu/esp8266: startup info if module esp_log_startup is used

### Contribution description

This PR realizes for the ESP8266 the same pseudomodule `esp_log_startup`, which was realized in PR #12810 for the ESP32. It is used to control the additional information that are printed at startup for ESP8266 boards.

Additional startup information, including the board configuration, are only printed when the `esp_log_startup` module is used. This reduces the amount of information that is printed by default to the console during the startup. The user can enable module `esp_log_startup` explicitly to get the additional startup information.

Module `esp_log_startup` is also enabled whith `LOG_LEVEL=4` or `LOG_LEVEL=5`.

### Testing procedure

Use any application to test the module with as following:

1. No additional information should be shown with the exception of the ROOM bootloader message.
    ```
    make BOARD=esp8266-esp-12x -C tests/thread_basic flash test
    ```
2. Additional information should be shown including the board configuration.
    ```
    LOG_LEVEL=4 make BOARD=esp8266-esp-12x -C tests/thread_basic/ test
    ```

### Issues/PRs references

#12810 is the same for ESP32
